test/unit/marked-spec.js

```diff
@@ -2,15 +2,29 @@ var marked = require('../../lib/marked.js');
 
 describe('Test heading ID functionality', function() {
   it('should add id attribute by default', function() {
-    var renderer = new marked.Renderer(marked.defaults);
-    var header = renderer.heading('test', 1, 'test');
-    expect(header).toBe('<h1 id="test">test</h1>\n');
+    var html = marked('# test');
+    expect(html).toBe('<h1 id="test">test</h1>\n');
+  });
+
+  it('should add unique id for repeating heading 1280', function() {
+    var html = marked('# test\n# test\n# test');
+    expect(html).toBe('<h1 id="test">test</h1>\n<h1 id="test-1">test</h1>\n<h1 id="test-2">test</h1>\n');
```

Comment:
unit tests should only test a single function (unit) not the whole application.

We should expose `Slugger` on marked so we can write tests like:

```suggestion
    var slugger = new marked.Slugger();
    expect(slugger.slug('test')).toBe('test');
    expect(slugger.slug('test')).toBe('test-1');
    expect(slugger.slug('test')).toBe('test-2');
```